### [성인 남성이 더 많이 죽었을 때 아이들도 남자아이가 더 많이 죽었을지]
어른중에서 남자와 여자중 누가 더 많이 죽었는지(corr(),비율 비교)  
아이 중에서 여자아이와 남자아이 생존 비교를 통해   
아이 중에서도 생존에 남녀의 차이가 있었는지
사용할 컬럼 : survived,sex,who

In [1]:
# [1] 모듈 로딩
import pandas as pd
import numpy as np

In [2]:
# [2] 데이터 준비
file_name=r'C:\Users\KDP-17\EX_PANDAS6\day_04\titanic.csv'


In [3]:
# [3] csv => DF
csvdf=pd.read_csv(file_name)

In [4]:
csvdf

survived  pclass     sex   age  sibsp  parch     fare embarked   class  \
0           0       3    male  22.0      1      0   7.2500        S   Third   
1           1       1  female  38.0      1      0  71.2833        C   First   
2           1       3  female  26.0      0      0   7.9250        S   Third   
3           1       1  female  35.0      1      0  53.1000        S   First   
4           0       3    male  35.0      0      0   8.0500        S   Third   
..        ...     ...     ...   ...    ...    ...      ...      ...     ...   
886         0       2    male  27.0      0      0  13.0000        S  Second   
887         1       1  female  19.0      0      0  30.0000        S   First   
888         0       3  female   NaN      1      2  23.4500        S   Third   
889         1       1    male  26.0      0      0  30.0000        C   First   
890         0       3    male  32.0      0      0   7.7500        Q   Third   

       who  adult_male deck  embark_town alive  alone  
0      man        True  NaN  Southampton    no  False  
1    woman       False    C    Cherbourg   yes  False  
2    woman       False  NaN  Southampton   yes   True  
3    woman       False    C  Southampton   yes  False  
4      man        True  NaN  Southampton    no   True  
..     ...         ...  ...          ...   ...    ...  
886    man        True  NaN  Southampton    no   True  
887  woman       False    B  Southampton   yes   True  
888  woman       False  NaN  Southampton    no  False  
889    man        True    C    Cherbourg   yes   True  
890    man        True  NaN   Queenstown    no   True  

[891 rows x 15 columns]

In [5]:
csvdf_data=csvdf[['survived','sex','who']]
csvdf_data

survived     sex    who
0           0    male    man
1           1  female  woman
2           1  female  woman
3           1  female  woman
4           0    male    man
..        ...     ...    ...
886         0    male    man
887         1  female  woman
888         0  female  woman
889         1    male    man
890         0    male    man

[891 rows x 3 columns]

In [6]:
# [4] 데이터 확인
csvdf_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 3 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   survived  891 non-null    int64 
 1   sex       891 non-null    object
 2   who       891 non-null    object
dtypes: int64(1), object(2)
memory usage: 21.0+ KB


In [7]:
# [5]
# [5-1] 결측치 체크
csvdf_data.isna().sum()


survived    0
sex         0
who         0
dtype: int64

In [8]:
# [5-2] 중복 데이터
# True, False 반환하는 값이 중복값으로 뜨는 거라서 중복 제거 할 필요 없음
csvdf_data.duplicated()

0      False
1      False
2       True
3       True
4       True
       ...  
886     True
887     True
888     True
889     True
890     True
Length: 891, dtype: bool

In [9]:
# [6-1] 탑승자 중 성인 찾기
csvdf_data_adult=(csvdf_data['who'] =='woman') | (csvdf_data['who'] =='man')
csvdf_data_adult.value_counts()

who
True     808
False     83
Name: count, dtype: int64

In [10]:
# [6-2] 탑승자 중 성인 여성 및 남성 찾기
csvdf_data_woman=(csvdf_data['who'] =='woman')
csvdf_data_woman.value_counts()
# 성인 여성 : 271명
# 성인 남성 : 537명

who
False    620
True     271
Name: count, dtype: int64

In [11]:
# [6-3] 탑승자 중 생존자 찾기
csvdf_data_suv=(csvdf_data['survived'] == 1)
csvdf_data_suv.value_counts()
# 생존자 : 342명

survived
False    549
True     342
Name: count, dtype: int64

In [12]:
# [6-4] 생존자 중 여성 찾기
csvdf_adult_suv=(csvdf_data['survived']==1) & (csvdf_data['who'] =='woman') 
csvdf_adult_suv.value_counts()
# 여성 성인 생존자 : 205명

False    686
True     205
Name: count, dtype: int64

In [13]:
# [6-5] 생존자 중 아이 찾기
csvdf_data_child=(csvdf_data['survived'] == 1) & (csvdf_data['who'] == 'child')
csvdf_data_child.value_counts()
# 생존한 아이 49명

False    842
True      49
Name: count, dtype: int64

탑승자 수 : 891  
탑승자 중 성인 수 : 808  
탑승자 중 아이 수 : 83  
생존자 수 : 342  
생존자 중 여성 성인 수 : 205 
생존자 중 남성 성인 수 : 88
생존자 중 아이 수 : 49

In [14]:
# [6-5] 탑승자 중 생존한 남성 성인, 여성 성인 비율
print(f'생존 남성 성인 비율:{88/808}, 생존 여성 성인 비율:{205/808}')

생존 남성 성인 비율:0.10891089108910891, 생존 여성 성인 비율:0.2537128712871287


In [15]:
# [6-6] 생존에 가장 영향을 많이 끼친 컬럼
csvdf.corr(numeric_only=True)


survived    pclass       age     sibsp     parch      fare  \
survived    1.000000 -0.338481 -0.077221 -0.035322  0.081629  0.257307   
pclass     -0.338481  1.000000 -0.369226  0.083081  0.018443 -0.549500   
age        -0.077221 -0.369226  1.000000 -0.308247 -0.189119  0.096067   
sibsp      -0.035322  0.083081 -0.308247  1.000000  0.414838  0.159651   
parch       0.081629  0.018443 -0.189119  0.414838  1.000000  0.216225   
fare        0.257307 -0.549500  0.096067  0.159651  0.216225  1.000000   
adult_male -0.557080  0.094035  0.280328 -0.253586 -0.349943 -0.182024   
alone      -0.203367  0.135207  0.198270 -0.584471 -0.583398 -0.271832   

            adult_male     alone  
survived     -0.557080 -0.203367  
pclass        0.094035  0.135207  
age           0.280328  0.198270  
sibsp        -0.253586 -0.584471  
parch        -0.349943 -0.583398  
fare         -0.182024 -0.271832  
adult_male    1.000000  0.404744  
alone         0.404744  1.000000

##### 성인 남성과 성인 여성의 생존 비율은 2배 이상 차이나며 생존에 가장 영향을 많이 끼친 컬럼은 성인 남성 여부라는 점을 알 수 있다

In [16]:
# [7-1] 탑승한 여아와 남아 찾기(애초에 탑승한 남아와 여아의 수가 차이 날 수 있어 계산함)
csvdf_data_boy=(csvdf_data['who'] == 'child') & (csvdf_data['sex'] == 'male')
csvdf_data_boy.value_counts()
# 남아 : 40
# 여아 : 43

False    851
True      40
Name: count, dtype: int64

In [17]:
# [7-2] 생존한 아이 중 여아와 남아 찾기
csvdf_data_boy2=(csvdf_data['survived'] == 1) & (csvdf_data['who'] == 'child') & (csvdf_data['sex'] == 'male')
csvdf_data_boy2.value_counts()
# 남아 : 21
# 여아 : 28

False    870
True      21
Name: count, dtype: int64

In [18]:
# [7-3] 탑승한 아이 중 생존한 여아와 남아의 비율
print(f'남아 비율 :{21/40}, 여아 비율 :{28/43}')

남아 비율 :0.525, 여아 비율 :0.6511627906976745


In [19]:
# [7-4] 생존한 아이 중 여아와 남아의 비율
print(f'남아 비율 :{21/49}, 여아 비율 :{28/49}')

남아 비율 :0.42857142857142855, 여아 비율 :0.5714285714285714


남아와 여아의 생존 비율은 성인만큼의 차이가 나지는 않지만 여아일 경우 생존할 확률이 남아 대비 15% 이상 되는 것을 확인할 수 있다